pkg/querier/queryrange/results_cache.go

```diff
@@ -171,7 +171,12 @@ func (s resultsCache) Do(ctx context.Context, r Request) (Response, error) {
 		response Response
 	)
 
-	maxCacheTime := int64(model.Now().Add(-s.cfg.MaxCacheFreshness))
+	// check if per-tenant cache freshness value is provided
+	maxCacheFreshness := s.limits.MaxCacheFreshness(userID)
+	if maxCacheFreshness == time.Duration(0) {
+		maxCacheFreshness = s.cfg.MaxCacheFreshness
+	}
```

Comment:
You have to do this because now there are two flags doing the same thing:
- `-frontend.per-user-max-cache-freshness` (new one, which allows overrides)
- `-frontend.max-cache-freshness` (old one)

I think this is not correct from a design perspective. We shouldn't introduce a new flag, but making the old one supporting overrides.